lib/schema/date.js

```diff
@@ -232,9 +232,12 @@ SchemaDate.prototype.cast = function(value) {
 
   if (value instanceof Number || typeof value === 'number') {
     date = new Date(value);
+  } else if (typeof value === 'string' && !isNaN(Number(value)) && (Number(value) >= 275761 || Number(value) < 0)) {
+    // string representation of milliseconds take this path
+    date = new Date(Number(value));
   } else if (typeof value.valueOf === 'function') {
-    // support for moment.js. This is also the path strings will take because strings
-    // have a `valueOf()`
+    // support for moment.js. This is also the path string representation of years
```

Comment:
Undo your changes to this comment please, because this is also the path that strings like `2017-06-01` take

Reply:
done